doc/src/devdocs/ast.md

```diff
@@ -763,3 +763,61 @@ Boolean properties:
     * 0x01 << 4 = the syntactic control flow within this method is guaranteed to terminate (`:terminates_locally`)
 
     See the documentation of `Base.@assume_effects` for more details.
+
+
+How to interpret line numbers in a `CodeInfo` object:
```

Comment:
Good questions. I will try to answer them inline, and eventually hopefully update the document to clarify these points also (maybe not immediate as I may be on vacation for the next 2 weeks)

Typically these are attached to a `CodeInfo` object in the `debuginfo` field, but they are expected to be referenced from many other places (one of the main points of this PR is that this will be referenced from many places instead of doing copies as much)